# HG changeset patch
# User Philip Nienhuis <prnienhuis@users.sf.net>
# Date 1428354341 -7200
#      Mon Apr 06 23:05:41 2015 +0200
# Node ID f68c3a62e42c8681687b986def23b0a993a7764f
# Parent  42c3a08293245811e022cad481d23034f3895e55
strread.m: properly process traling delimiters in case of
           MultipleDelimsAsOne parameter (bug #44750)

diff --git a/scripts/io/strread.m b/scripts/io/strread.m
--- a/scripts/io/strread.m
+++ b/scripts/io/strread.m
@@ -800,17 +800,17 @@ function out = split_by (text, sep, mult
   else
     mult_dlms_s1 = false;
   endif
 
   ## Split text string along delimiters
   out = ostrsplit (text, sep, mult_dlms_s1);
   if (index (sep, eol_char)); out = strrep (out, char (255), ''); endif
   ## In case of trailing delimiter, strip stray last empty word
-  if (! isempty (out) && any (sep == text(end)))
+  if (! isempty (out) && any (sep == text(end)) && ! mult_dlms_s1)
     out(end) = [];
   endif
 
   ## Empty cells converted to empty cellstrings.
   out(cellfun ("isempty", out)) = {""};
 
 endfunction
 
@@ -932,16 +932,21 @@ endfunction
 %!test
 %! str = "11, 12, 13,, 15\n21,, 23, 24, 25\n,, 33, 34, 35";
 %! [a b c d] = strread (str, "%f %f %f %f", "delimiter", ",", "multipledelimsasone", 1, "endofline", "\n");
 %! assert (a', [11, 21, NaN]);
 %! assert (b', [12, 23, 33]);
 %! assert (c', [13, 24, 34]);
 %! assert (d', [15, 25, 35]);
 
+## Bug #44750
+%!test
+%! assert (strread ('/home/foo/','%s','delimiter','/','MultipleDelimsAsOne',1), ...
+%!         {"home"; "foo"});
+
 ## delimiter as sq_string and dq_string
 %!test
 %! assert (strread ("1\n2\n3", "%d", "delimiter", "\n"),
 %!         strread ("1\n2\n3", "%d", "delimiter", '\n'))
 
 ## whitespace as sq_string and dq_string
 %!test
 %! assert (strread ("1\b2\r3\b4\t5", "%d", "whitespace", "\b\r\n\t"),
